caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditEventBuilder.java

```diff
@@ -0,0 +1,273 @@
+    }
+
+    @Override
+    public void addEventParameter(String name, String columnName, boolean value) {
+        auditEventParams.add(new EventParam(name, "boolean", columnName, value));
+    }
+
+    @Override
+    public void addEventParameter(String name, String columnName, Date value) {
+        auditEventParams.add(new EventParam(name, "date", columnName, value));
+    }
+
+    @Override
+    public void send() throws Exception {
+
+        //  Get the constructed Audit Event as a JSON string
+        String auditEventJson = getAuditEventAsJsonString();
+
+        // Reset the JsonBuilder
+        jsonBuilder = XContentFactory.jsonBuilder();
+
+        //  An Audit Event in Json format was not returned, throw exception
+        if (auditEventJson == null || auditEventJson.isEmpty()) {
+            String errorMessage = "No Audit Event JSON to send to the Audit webservice";
+            LOG.error(errorMessage);
+            throw new WebserviceClientException(errorMessage);
+        }
+
+        URL url = new URL("http://" + webserviceHostAndPort + "/caf-audit-service/v1/auditevents");
+        HttpURLConnection conn = (HttpURLConnection) url.openConnection();
+        //  Try to send Audit Event to the Audit Webservice HTTP endpoint
+        try {
+            conn.setConnectTimeout(5000);
+            conn.setRequestProperty("Content-Type", "application/json; charset=UTF-8");
+            conn.setDoOutput(true);
+
+            byte[] auditEventJsonBytes = auditEventJson.getBytes("UTF-8");
+
+            //  The number of bytes to send is known; set fixed length streaming mode
+            conn.setFixedLengthStreamingMode(auditEventJsonBytes.length);
+
+            OutputStream outputStream = conn.getOutputStream();
+            outputStream.write(auditEventJsonBytes);
+            outputStream.close();
+
+            int responseCode = conn.getResponseCode();
+
+            // Check that the response code was returned as expected. Print a WARN if the response code was a 200 but
+            // not 204. Print an error if the response code is anything else.
+            int expectedResponseCode = 204;
+            if (responseCode >= 200 && responseCode < 300) {
+                if (responseCode != expectedResponseCode) {
+                    LOG.warn("Webservice accepted request but returned response code " + responseCode
+                            + " when the expected response code is " + expectedResponseCode);
+                }
+            } else {
+                String errorMessage = "Webservice returned response code " + responseCode + " when the expected " +
+                        "response code is " + expectedResponseCode;
+                LOG.error(errorMessage);
+                throw new WebserviceClientException(errorMessage);
+            }
+            LOG.info("Audit event request sent and received response code " + responseCode + " from the webservice");
+        } finally {
```

Comment:
As mentioned above, this method need try-with-resource and/or try/catch blocks.  When re-worked this may affect the content of the finally block.